x-pack/filebeat/module/kubernetes/_meta/fields.yml

```diff
@@ -0,0 +1,134 @@
+- key: kubernetes
+  title: Kubernetes
+  description: >
+    Module for handling logs produced by Kubenetes pods
+  fields:
+  - name: coredns
+    type: group
+    description: >
+      Fields from coredns logs after normalization
+    fields:
+    - name: duration
+      type: double
+
+    - name: client
+      type: ip
+
+    - name: size
+      type: integer
+
+    - name: dnssec_ok
+      type: boolean
+
+    - name: proto
+      type: keyword 
+
+    - name: qclass
+      type: keyword
+
+    - name: rsize
+      type: integer
+
+    - name: log_level
+      type: keyword
+
+    - name: id
+      type: keyword
+
+    - name: rcode
+      type: keyword
+
+    - name: port
+      type: integer
+
+    - name: bufsize
+      type: integer
+
+    - name: qtype
+      type: keyword
+
+    - name: rflags
+      type: keyword
+      description: >
+        coredns response flags
+
+    - name: qname
+      type: keyword
+      description: >
+        coredns query name
+
+    - name: message
+      type: text
+      description: >
+        coredns original log message
+
+  - name: kubernetes
+    type: group
+    description: >
+      Kubernetes meta fields
+    fields:
+    - name: replicaset
+      type: group
+      fields:
+      - name: name
+        type: keyword
+
+  - name: envoy
+    type: group
+    description: >
+      Fields from envoy proxy logs after normalization
+    fields:
+    - name: log_type
+      type: keyword
+      description: >
+        Envoy log type, normally ACCESS
+
+    - name: method
+      type: keyword
+
+    - name: path
+      type: keyword
+
+    - name: proto 
+      type: keyword
+
+    - name: response_code
+      type: integer
+
+    - name: response_flags
+      type: keyword
+
+    - name: bytes_received
+      type: long
+
+    - name: bytes_sent
+      type: long
+
+    - name: duration
+      type: double 
+
+    - name: upstream_service_time
+      type: double
+
+    - name: forwarded_for
+      type: keyword
+
+    - name: http_user_agent
```

Comment:
ECS `user_agent.original`

Reply:
In general, all the related ECS fields are added via the beats or ingest pipeline processors. The original fields are kept for reference. I guess I should remove these original fields instead.

Reply:
Yes. The team has been going back and forth on this.

For all plaintext logs, the cutover has been pretty straightforward and drastic. We parse right into the ECS fields as much as we can, and add custom fields for everything not covered.

For Suricata, we started with the assumption that some pre-existing content or community experience existed, based on the exact Suricata field names & so on, because of the existence of their EVE JSON log format. I'm not sure if that's the case or not.

But since 7.0, we've started "hollowing out" the `suricata.eve.*` namespace, and replacing all of the fields that map to ECS with field aliases anyway ([examples](https://github.com/elastic/beats/blob/7.0/x-pack/filebeat/module/suricata/eve/_meta/fields.yml#L70-L94)).

So I think we should do the same here. Especially since this is a brand new module. Let's not create legacy we'll have to clean up in 8.0 ;-)